### ISSUE TITLE
Fix Request: false positive in AvoidImplicitlyRecompilingRegex for Kotlin

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#615]
regex is a parameter, variable:
```kotlin
    fun storeUsingRegex(regex: String, sourceName: String, dataTable: DataTable) {

        val dtm = dataTable.asMaps()
        val source = testContext.parameter(sourceName)
        val matches = Regex(regex).findAll(source.toString()).toList()
```
